xdsl/dialects/builtin.py

```diff
@@ -1891,8 +1891,8 @@ def get_resolvers(
         }
 
     def verify(self, attr: Attribute, constraint_context: ConstraintContext) -> None:
-        if isinstance(attr, VectorType) or isinstance(attr, TensorType):
-            attr = cast(VectorType[Attribute] | TensorType[Attribute], attr)
+        if isinstance(attr, MemRefType) or isinstance(attr, TensorType):
+            attr = cast(MemRefType[Attribute] | TensorType[Attribute], attr)
```

Comment:
Would you mind moving this to a separate PR with a test case?
We seem to be lacking any verification testing on this type.

Reply:
Seeing the changes, I'd still recommend doing this.

Reply:
I will do that in the next few days